### PR TITLE
feat: Add RandomJPEG Augmentation

### PR DESCRIPTION
**Pull request by ChristophReich1996**
#### Changes
This pull request wraps the differentiable JPEG implementation as an augmentation module. In particular, the random JPEG augmentation samples (uniformly) a JPEG quality (compression rate) and applies differentiable JPEG coding. Tests for augmentation were also implemented.

#### Type of change

- [x] 📚  Documentation Update
- [x] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📝 This change requires a documentation update

### CODE REVIEW

**Review by edgarriba — APPROVED** [review]

File: kornia/augmentation/_2d/intensity/jpeg.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+from kornia.augmentation import random_generator as rg
+from kornia.augmentation._2d.intensity.base import IntensityAugmentationBase2D
+from kornia.core import Tensor
+from kornia.enhance import jpeg_codec_differentiable
+
+
+class RandomJPEG(IntensityAugmentationBase2D):
+    r"""Applies random (differentiable) JPEG coding to a tensor image.
+
```
Comment (shijianjian) [inline kornia/augmentation/_2d/intensity/jpeg.py:11]: Can you add an image example here?

**Review by shijianjian — APPROVED** [review]

**Review by johnnv1 — APPROVED** [review]

File: kornia/augmentation/_2d/intensity/jpeg.py (hunk 1 above)
Comment (ChristophReich1996) [reply kornia/augmentation/_2d/intensity/jpeg.py:11]: Can we use the same image as here? 😅

https://github.com/kornia/kornia/blob/e40da282ba84b02ed8f3958dc220ce76d20572aa/kornia/enhance/jpeg.py#L410

Comment (edgarriba) [reply kornia/augmentation/_2d/intensity/jpeg.py:11]: ideally we auto generate here https://github.com/kornia/kornia/blob/main/docs/generate_examples.py

Comment (ChristophReich1996) [reply kornia/augmentation/_2d/intensity/jpeg.py:11]: Just added `RandomJPEG` to the auto generate script and linked the resulting image.

**Pull request merged**